providers/stores/file.js

```diff
@@ -138,14 +151,19 @@ class FileStore extends AbstractStore {
 
   async store(coordinates, stream) {
     const filePath = this._toStoragePathFromCoordinates(coordinates) + '.json'
+    const preservedName = EntityCoordinates.fromObject(coordinates).toString()
     const dirName = path.dirname(filePath)
     await promisify(mkdirp)(dirName)
     return new Promise((resolve, reject) => {
-      const file = fs
-        .createWriteStream(filePath)
-        .on('finish', () => resolve())
+      const fileStream = fs.createWriteStream(filePath).on('error', error => reject(error))
+      const jsonStream = JSONStream.parse()
+        .on('data', data => {
+          data._metadata.coordinates = preservedName
+          fileStream.write(JSON.stringify(data))
+        })
+        .on('end', () => resolve())
```

Comment:
should we be resolving when the fileStream `finish`es? Seems like the JSONstream will `end` when `stream` is done piping to it but we may still not have written the file?

Reply:
I will double check on that

Reply:
The more I looked into this, I realized I was doing it wrong and ended up dropping JSONStream altogether. Now we will do the following:

1. read the input stream to the end
1. parse the JSON
1. modify the JSON
1. write the JSON to disk
1. resolve the promise